### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by Rythmag**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,49 @@ We used elasticsearch as our database engine. We utilise elasticsearch's inbuilt
...
+    sudo apt-get install maven
+To run the app use:
+
+    mvn package appengine:run
+##### Java 8:
+**Installing Java 8**:
+
+    sudo apt-get install default-jdk
+    sudo apt-get install oracle-java8-installer
+**Setiing Java variables**:
```
Comment (shudhatmajain) [inline README.md:33]: Setting

File: README.md [hunk 2]
```diff
@@ -18,6 +18,49 @@ We used elasticsearch as our database engine. We utilise elasticsearch's inbuilt
 - **Server:**
 The database is connected to the server using java high-level REST API. Maven app engine is used to run the application.
 
+### Prerequisite
+##### Maven:
+
```
Comment (shudhatmajain) [inline README.md:23]: Installing maven.

File: README.md [hunk 3]
```diff
@@ -18,6 +18,49 @@ We used elasticsearch as our database engine. We utilise elasticsearch's inbuilt
...
+    JAVA_HOME="/usr/lib/jvm/java-8-oracle"
+Save and exit the file, and reload it
+
+    source /etc/environment
+ ##### Elasticsearch:
+ 
+[Elasticsearch installation link](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/install-elasticsearch.html)
+
+##### Kibana:
```
Comment (shudhatmajain) [inline README.md:50]: Can we add details about where to put IP address for running ElasticSearch in the code.

File: README.md [hunk 4]
```diff
@@ -18,6 +18,49 @@ We used elasticsearch as our database engine. We utilise elasticsearch's inbuilt
...
+
+[Kibana installation link](https://www.elastic.co/guide/en/kibana/7.x/install.html)
+
+##### Google Custom Search API:
```
Comment (shudhatmajain) [inline README.md:54]: Can we add details about where to put API key for Search API.

File: README.md [hunk 5]
```diff
@@ -18,6 +18,49 @@ We used elasticsearch as our database engine. We utilise elasticsearch's inbuilt
...
+
+[CustomSearch API](https://developers.google.com/custom-search/)
+
+##### Index template Setting
+
+Set the index template given in
+error-analyzer-and-fix-suggester/src/main/java/com/google/error_analyzer/backend/index_template.json 
+using kibana. For setting index template refer [Index template setting](https://www.elastic.co/guide/en/elasticsearch/reference/7.x/indices-templates.html)
+
```
Comment (nishantnahata) [inline README.md:63]: Section for turning up the server.

**Pull request merged**